src/v/kafka/server/handlers/describe_producers.cc

```diff
@@ -0,0 +1,180 @@
+                klog.error,
+                "Can not get rm_stm for ntp({}). Looks like transaction are "
+                "disabled",
+                ntp);
+
+              return ss::make_ready_future<ret_res>(
+                kafka::error_code::unknown_server_error);
+          }
+
+          return rm_stm_ptr->get_transactions().then(
+            [ntp](auto tx) -> ss::future<ret_res> {
+                if (!tx.has_value()) {
+                    if (tx.error() == cluster::errc::not_leader) {
+                        return ss::make_ready_future<ret_res>(
+                          kafka::error_code::not_leader_for_partition);
+                    }
+
+                    vlog(
+                      klog.warn,
+                      "get_transactions returned unexpected error: {}",
+                      tx.error());
+                    return ss::make_ready_future<ret_res>(
+                      kafka::error_code::unknown_server_error);
+                }
+
+                return ss::make_ready_future<ret_res>(tx.value());
+            });
+      });
+}
+
+} // namespace
+
+template<>
+ss::future<response_ptr>
+describe_producers_handler::handle(request_context ctx, ss::smp_service_group) {
+    describe_producers_request request;
+    request.decode(ctx.reader(), ctx.header().version);
+    log_request(ctx.header(), request);
+
+    describe_producers_response response;
+
+    auto unauthorized_it = std::partition(
+      request.data.topics.begin(),
+      request.data.topics.end(),
+      [&ctx](const topic_request& topic) {
+          return ctx.authorized(security::acl_operation::read, topic.name);
+      });
+
+    std::vector<topic_request> unauthorized(
+      std::make_move_iterator(unauthorized_it),
+      std::make_move_iterator(request.data.topics.end()));
+
+    request.data.topics.erase(unauthorized_it, request.data.topics.end());
+
+    for (const auto& topic : request.data.topics) {
+        topic_response topic_resp;
+        topic_resp.name = topic.name;
+        for (const auto& partition : topic.partition_indexes) {
+            partition_response partition_resp;
+            partition_resp.partition_index = partition;
+            model::ntp ntp(model::kafka_namespace, topic.name, partition);
+            auto txs = co_await get_producers_for_partition(ctx, ntp);
+            if (txs.has_value()) {
+                for (const auto& tx : txs.value()) {
+                    producer_state producer_info;
+                    producer_info.producer_epoch = tx.first.epoch;
+                    producer_info.producer_id = tx.first.id;
+
+                    if (tx.second.info) {
+                        producer_info.last_timestamp
+                          = convert_rm_stm_time_to_milliseconds(
+                            tx.second.info->last_update);
+                    }
+
+                    // TODO: We do not store coordinator epoch
+                    producer_info.coordinator_epoch = -1;
```

Comment:
is this feature will be complete without the information about coordinator epoch ? Is it used by the clients in a way that that it would stop them from making progress ?

Reply:
Yes. It will work. We are gonna handleAbort tx request, so we gonna ignore some fields from client.
If not - I will fix it in next pr